packages/squiggle-lang/src/XYShape.ts

```diff
@@ -640,3 +665,127 @@ export const Analysis = {
     return meanOfSquares - meanSquared;
   },
 };
+
+type XYShapeSubset = {
+  points: [number, number][];
+  segments: XYShape[];
+};
+
+type ComparisonType =
+  | "greaterThan"
+  | "lesserThan"
+  | "equals"
+  | "greaterThanOrEqual"
+  | "lessThanOrEqual";
+
+/**
+  This function extracts a subset of the input shape that satisfies the comparison condition with the threshold.
+  It uses linear interpolation to find the exact points where the condition is satisfied.
+  Right now, we only use the `greaterThan` comparison type, for getting the Support of continuous distributions, but we can improve this in the future.
+  It assumes that the input shape is sorted, has no duplicate x values, and uses linear interpolation.
+  Note that if you want to use the result as a distribution, you will have to combine the segments somehow. They don't have surrounding points at 0 yet, so you can't just add them together.
+*/
+export const extractSubsetThatSatisfiesThreshold = (
+  shape: XYShape,
+  comparisonType: ComparisonType,
+  threshold: number
+): XYShapeSubset => {
+  const comparisonIsSatisfied = (y: number, _threshold: number): boolean => {
+    switch (comparisonType) {
+      case "greaterThan":
+        return y > _threshold;
+      case "lesserThan":
+        return y < _threshold;
+      case "equals":
+        return Math.abs(y - _threshold) < Number.EPSILON;
```

Comment:
There are probably subtle bugs here, or, at least, this boundary is arbitrary; `Number.EPSILON` is not the smallest representable number.

Not sure if this is important.